runner-review.sh

```diff
@@ -0,0 +1,333 @@
+SHORT_OPTIONS=c:
+LONG_OPTIONS=failed-count:,net-trace,sys-trace,call-trace,log-trace,debug
+
+OPTS=$(getopt --options=${SHORT_OPTIONS} --longoptions=${LONG_OPTIONS} --name "$0" -- "$@")
+if [[ $? -ne 0 ]]; then
+    # e.g. $? == 1
+    #  then getopt has complained about wrong arguments to stdout
+    exit 2
+fi
+
+
+# read getopt's output this way to handle the quoting right:
+eval set -- "${OPTS}"
+
+
+
+# parse each options from argument list until getting separate --
+while true; do
+    case "$1" in
+	
+	    -c)
+	    # Getting & store the execution count/times
+            export COUNT="$2"
+            shift 2
+            ;;
+			
+        --failed-count)
+            # Getting & store the failed execution count
+            export FAIL_COUNT="$2"
+            shift 2
+            ;;
+			
+        --debug)
+	    # Setting the debug options
+            export DEBUG="set -xvf"
+            shift
+            ;;
+			
+        --net-trace)
+	    # trigger tcpdump to generate network traffic : we used - /usr/sbin/tcpdump
+            export NET_TRACE="Y"
+            shift
+            ;;
+        --sys-trace)
+	    # triggering OPVA Extract  utility to generate system resources measured matrices : we used /opt/perf/bin/extract
+            export SYS_TRACE="Y"
+            shift
+            ;;
+			
+        --call-trace)
+	    # trigger the strace to generate call trace - we used /usr/bin/strace 
+            export CALL_TRACE="Y"
+            shift
+            ;;
+			
+        --log-trace)
+	    # trigger the log trace by printing the stdout & stderr to separate log files 
+            export LOG_TRACE="Y"
+            export LOG_OUT="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.out"
+            export LOG_ERR="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.err"
+            shift
+            ;;
+
+
+        --)
+		# once we arrived to -- this end of arguments parsing 
+            shift
+            break
+            ;;
+        *)
+		# when the input from OPTS no matching any of above , then it means the input not valid/or something wrong happened 
+            usage
+            ;;
+    esac
+done
+
+
+# enable/disable the debug option depend on the input options (if --debug was passed)
+${DEBUG}
+
+
+
+
+# handle non-option arguments , check if the input argument(not for the OPTS) mount is only 1 argument(one argument for the command parameter)
+if [[ $# -ne 1 ]]; then
+    usage
+fi
+
+
+# getting the command to run from command-line argument 
+COMMAND="$1"
+
+# declare array to store all the returned value for each command execution
+set -a list_rc
+
+
+
+
+
+
+
+# command executions loop - main invocations steps 
+# Test and check the executions time and allowed fails counters 
+while [[ ${execution_count} != ${COUNT} ]] && [[ ${failed_executions} != ${FAIL_COUNT} ]] ; do
+
+      export START_DATE=$(date +"%m/%d/%Y %H:%M:%S")
+      export CALL_TRACE_OUT="${OUTPUT_DIR}/call_trace_`${TIME_SIGNUTURE}`.out"
```

Comment:
If `strace` will be called more than once in less than a second, the log will be overridden.
You can utilize `mktemp` for this task.

Reply:
For the files allocation/creation decision - i thought about two ideas 
define a file with timestamp as it wil
`mktemp` util could be used here and utilized by 
i thought about this util `mktemp` as it will create the files under `/tmp` (by default, when `$TMPDIR` not set) and it used for a file size is few KB's , but in case the `starce` output file size >KB's then it will overload the filesystem /tmp and will affect the OS system and other programs .
in other side , `mktemp` could be